src/haxeLanguageServer/features/haxe/codeAction/DiagnosticsCodeActionFeature.hx

```diff
@@ -203,9 +210,164 @@ class DiagnosticsCodeActionFeature implements CodeActionContributor {
 			});
 		}
 
+		final tooManyArgsRe = ~/Too many arguments([\w.]*)/;
+		if (tooManyArgsRe.match(arg)) {
+			final document = context.documents.getHaxe(params.textDocument.uri);
+			final replacement = document.getText(diagnostic.range);
+			actions.push({
+				title: "Add argument",
+				data: CodeActionFeature.addResolveData(MissingArg(action -> createMissingArgumentsAction(action, params, diagnostic))),
```

Comment:
I think `data` should just be `{textDocument: params.textDocument, offset: diagnostic.range.start}` 
that would eliminate the need to maintain a separate data structure with callbacks

of course `onCodeActionResolve` would need to be able to differentiate between different data payloads if / when we add more code actions that use resolving

Reply:
Then that function should be static to be called from different class, and you need more things like `context` as arguments. I made `action.data` optionally typed now, but i think such enum with callback is fine for more code actions, easy to extend in comparison to one big object with fields from different actions?